server/api/chef_handler.py

```diff
@@ -0,0 +1,63 @@
+from models.User import User, UserSchema
+from flask import request, Response, jsonify
+from flask_restful import Resource
+from flask_jwt_extended import (
+    create_access_token,
+    set_access_cookies,
+    create_refresh_token,
+    get_csrf_token,
+    set_refresh_cookies
+)
+from helpers.api import custom_json_response
+from helpers.database import save_to_database
+from helpers.distance import distance
+from sqlalchemy import exc
+
+user_schema = UserSchema()
+user_schema_private = UserSchema(exclude=['password', 'email', 'isChef', 'address'])
+
+
+class ChefResource(Resource):
+    def get(self, id=None):
+        q_params = request.args
+        if id:
+            user = User.get_by_id(id)
+            return user_schema_private.dump(user)
+
+        filters = []
+        all_chefs = User.get_all_chefs()
+        ser_chefs = user_schema_private.dump(all_chefs, many=True)
+        
+        # stub data
+        chef_loc = (43.643523, -79.386722)  # CN Tower, Toronto
+        cuisines_str = "Japanese, Chinese, Mexican"
+        chef_cuisines = [c.strip() for c in cuisines_str.split(",")]
+        # userLoc = (43.664389, -79.392249)  # Queens PArk, Toronto
+
+        # supply maxDistance, userLat, and userLong to filter by distance
+        if (q_params.get("maxDistance") and q_params.get("userLat") and q_params.get("userLon")):
+            user_loc = (float(q_params.get("userLat")), float(q_params.get("userLon")))
+            max_dist = float(q_params.get("maxDistance"))
+            filters.append("distance")
+            ser_chefs = [
+                chef for chef in ser_chefs
+                if distance(chef_loc, user_loc) < max_dist
+            ]
```

Comment:
Instead of getting all users and then doing filtering... we can actually more ideally do filtering in SQL. You can filter using a formula and find chefs using their cuisines. At the very least, we should be doing the fitlering for chefs by cuisine in a SQL call. This would be more efficient when our database has a lot of chefs

Reply:
Noted on this. Just wanted to get this to work while the cuisines table has not been implemented yet.